### PR TITLE
Automatic download of the UKPN Grafana data 

### PR DESCRIPTION
**Pull request by vrym2**
# Pull Request

## Description
This PR corresponds to automatic data acquisition from UKPN `Grafana` dashboard. For the weblink of the dashboard, please click [here](https://dsodashboard.ukpowernetworks.co.uk/)
Fixes #

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

- [ ] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -127,9 +127,15 @@ dmypy.json
...
+tests/scripts/data
+tests/data/
 pv-solar-farm-forecasting.code-workspace
 .vscode
 .gitattributes
+.flake8
+.pre-commit-config.yaml
+environment.yml
+pydoc-markdown.yml
+requirements.txt
```
Comment (jacobbieker) [inline .gitignore:139]: You shouldn't ignore this and the environment.yml, or pre commit config, as they are used for building and linting

File: tests/grafana/test_grafana.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        "ninfield",
+        "northfleet_east",
+        "norwich",
+        "rayleigh",
+        "richborough",
+        "sellindge"]
+
+    gsp_names = list(reversed(get_gsp_names()))
+    download_directory = (
+        "/home/vardh/ocf/pv-solar-farm-forecasting/tests/data/grafana_dashboard_ukpn"
```
Comment (jacobbieker) [inline tests/grafana/test_grafana.py:60]: Don't hardcode the local path

**Review by jacobbieker — COMMENTED** [review]
A few small changes, but looks good to me overall

**Comment by jacobbieker** [comment]
Just make sure all the tests pass before merging

File: .gitignore (hunk 1 above)
Comment (vrym2) [reply .gitignore]: yes, I updated this!

File: tests/grafana/test_grafana.py (hunk 2 above)
Comment (vrym2) [reply tests/grafana/test_grafana.py]: yes, testing is also unsuccessful, cause it is trying to check if the files are getting downloaded, and I'm not uploading raw csv files into git, I have changed the test code to just check if csv files are downloadable!

**Pull request merged**